src/rabbit_queue_index.erl

```diff
@@ -508,14 +508,18 @@ erase_index_dir(Dir) ->
     end.
 
 blank_state(QueueName) ->
-    blank_state_dir(
-      filename:join(queues_dir(), queue_name_to_dir_name(QueueName))).
+    blank_state_dir(queue_dir(QueueName)).
 
 blank_state_dir(Dir) ->
     blank_state_dir_funs(Dir,
                          fun (_) -> ok end,
                          fun (_) -> ok end).
 
+queue_dir(#resource{ virtual_host = VHost } = QueueName) ->
+    %% Queue directory is rabbit_mnesia_dir/:vhost/queues/:queue_id
```

Comment:
It would be a bit clearer if we used `{node_database_dir}/vhosts/{vhost}/queues/{queue}`.